### DISCUSSION TITLE
Missing text and strangely clipped elements after RmlUi 6.0 upgrade

### DISCUSSION THREAD

**Discussion by s1sw** [mikke89/RmlUi#677] [Q&A]
I recently upgraded my application to RmlUi 6.0, and I've been seeing some strange rendering issues - namely incorrectly clipped elements, and missing text. The problems are strangely non-deterministic in that they only seem to occur once the document has been loaded into a new context after a previous one containing that document has been destroyed. Checking render captures, RmlUi seems to be passing incorrect scissor rectangles to the render backend, and also not submitting any draws for the missing text (potentially some kind of culling going wrong?). I'm not confident enough that the problem isn't in my own code for an issue or similar, so any ideas on what could be going on are appreciated.

Missing text, both in a context rendering to an offscreen target and in a fullscreen context (element info panel of the debugger and box in bottom right corner):
![image](https://github.com/user-attachments/assets/0ecea932-3dea-48b6-b333-82982bc73063)

and the intended display (before the contexts are recreated):
![image](https://github.com/user-attachments/assets/c4715190-1914-46e9-8524-4444cce3f74e)

RenderDoc capture showing incorrectly clipped elements:
![image](https://github.com/user-attachments/assets/4f7b17f5-9563-44b2-a120-4567a66367ff)

and the correctly clipped elements (before context recreation):
![image](https://github.com/user-attachments/assets/cb86aea8-6eaf-4304-8809-97d3791b3cd3)

**Comment by mikke89** [comment]
Are you using our bult-in backends?

We do clamp the scissor region to the size of the context. It could perhaps look like it's clamping to the wrong size here, for some reason.

Are you able to track down how the scissor region is clamped and set? From a debugger, you can try with a set point in the  render interface scissor region, and work back from there.

**Reply by s1sw** [reply]
Thanks for the suggestion! I was able to track down the scissor region being clamped to the current viewport dimensions in `RenderManager`, where I found that `SetViewport` was only being called from `Context::SetDimensions`. Since I'm using multiple contexts, the viewport dimension would stay stuck to whichever context was created/resized last, causing the inconsistent clipping. 
As a workaround, I added a `SetViewport` call to `Context::Render` which has fixed the issue.

For what it's worth, this is on a custom Vulkan backend, but I'm pretty sure this will manifest on any backend when using multiple differently sized contexts.

**Reply by s1sw** [reply]
Made a PR for a fix in #680.
